matchers/be_a_file_matching_test.go

```diff
@@ -0,0 +1,157 @@
+package matchers_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/occam/matchers"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testBeAFileMatching(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect   = NewWithT(t).Expect
+		filename string
+		matcher  *matchers.BeAFileMatchingMatcher
+	)
+
+	it.Before(func() {
+		filename = filepath.Join(t.TempDir(), "file")
+
+		Expect(os.WriteFile(filename, []byte("hello world"), os.ModePerm)).To(Succeed())
+	})
+
+	context("will wrap ContainsSubstring", func() {
+		it("will pass", func() {
+			matcher = matchers.BeAFileMatching(ContainSubstring("hello"))
+
+			match, err := matcher.Match(filename)
+			Expect(match).To(Equal(true))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it("will fail", func() {
+			matcher = matchers.BeAFileMatching(ContainSubstring("foobar"))
+
+			match, err := matcher.Match(filename)
+			Expect(match).To(Equal(false))
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	context("will wrap Equal", func() {
+		it("will pass", func() {
+			matcher = matchers.BeAFileMatching(Equal("hello world"))
+
+			match, err := matcher.Match(filename)
+			Expect(match).To(Equal(true))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it("will fail", func() {
+			matcher = matchers.BeAFileMatching(Equal("foobar"))
+
+			match, err := matcher.Match(filename)
+			Expect(match).To(Equal(false))
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
```

Comment:
Would it be possible to have this be the default? I'd love to see a short-hand like the following:

```go
Expect("/tmp/some-path").To(BeAFileMatching("some-contents"))
```

Reply:
Good idea. Done.